### PR TITLE
DT-1020 added csv validation logic, made tests

### PR DESCRIPTION
**Pull request by joejacob**
added all csv logic and made some tests

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,59 @@
+PATH
```
Comment (isaacanthony) [inline Gemfile.lock]: This needs to be removed. I think you may have to take out the space in the .gitignore before the Gemfile.lock line. Same with .ruby-version, .ruby-gemset, and coverage lines. Then rm this file and git rm it.

File: filetype_validation.gemspec [hunk 2]
```diff
@@ -37,4 +37,6 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rake', '~> 10.0'
   spec.add_development_dependency 'rspec', '~> 3.0'
   spec.add_development_dependency 'rubocop', '~> 0.40.0'
+  spec.add_development_dependency 'ptools'
```
Comment (isaacanthony) [inline filetype_validation.gemspec]: We don't need ptools anymore.

File: lib/filetype_validation/csv_validator.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      csv_extension? && parse_csv?
+    end
+
+    private
+
+    #
+    # Simply checks if the file has a .csv extension
+    #
+    def csv_extension?
+      '.csv'.casecmp(File.extname(file.path))
```
Comment (isaacanthony) [inline lib/filetype_validation/csv_validator.rb]: Nice. :+1:

File: lib/filetype_validation/csv_validator.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    #
+    # This method checks to see if the file can have its
+    # first number_of_lines parsed as a csv file using the CSV
+    # library.
+    #
+    def parse_csv?
+      file_sample = File.foreach(file.path)
+                        .first(@options[:max_lines] || MAX_LINES)
+                        .join
+      CSV.parse(file_sample, headers: true)
```
Comment (isaacanthony) [inline lib/filetype_validation/csv_validator.rb]: Maybe we should allow an options for headers true/false like we do with max_lines. Then also have a HEADERS constant where our default is true.

File: spec/csv_validator_spec.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+require 'tempfile'
+
+describe FiletypeValidation::CsvValidator do
+  let(:result) { FiletypeValidation.csv?(file.flush, options) }
```
Comment (isaacanthony) [inline spec/csv_validator_spec.rb]: Why do you flush the file? Probably the right thing to do, I'm just not sure what it does.

File: spec/csv_validator_spec.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+      it { expect(result).to be(false) }
+    end
+  end
+
+  context 'checking csv parsing' do
+    let(:file) { Tempfile.new(['test', '.csv']) << file_content }
+
+    context 'returns true if the file can be parsed as a csv' do
+      let(:file_content) do
+        'street,city,zip,state,beds,baths,sq__ft,type
```
Comment (isaacanthony) [inline spec/csv_validator_spec.rb]: Ruby doesn't do well with multiline strings. Do 'first line' \ 'second line' instead.

File: spec/filetype_validation_spec.rb [hunk 7]
```diff
@@ -1,7 +0,0 @@
-require 'spec_helper'
```
Comment (isaacanthony) [inline spec/filetype_validation_spec.rb]: We can keep this test. We could just put it under spec/filetype_validation_spec/validation_spec.rb.

File: spec/csv_validator_spec.rb (hunk 5 above)
Comment (joejacob) [inline spec/csv_validator_spec.rb]: it's so i can read the file from the beginning after writing to it

Comment (isaacanthony) [inline spec/csv_validator_spec.rb]: Oh, ok. Awesome. :+1:

File: lib/filetype_validation/csv_validator.rb [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    end
+
+    #
+    # This method checks to see if the file can have its
+    # first number_of_lines parsed as a csv file using the CSV
+    # library.
+    #
+    def parse_csv?
+      file_sample = File.foreach(file.path)
+                        .first(@options[:max_lines] || MAX_LINES)
```
Comment (isaacanthony) [inline lib/filetype_validation/csv_validator.rb]: We define attr_readers for these. Lets use them then. options vs. @options.

File: spec/csv_validator_spec.rb [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    end
+  end
+
+  context 'checking csv parsing' do
+    let(:file) { Tempfile.new(['test', '.csv']) << file_content }
+
+    context 'returns true if the file can be parsed as a csv' do
+      let(:file_content) do
+        'street,city,zip,state,beds,baths,sq__ft,type' \
+        '\n3526 HIGH ST,SACRAMENTO,95838,CA,2,1,836,Residential'
```
Comment (isaacanthony) [inline spec/csv_validator_spec.rb]: This needs to be double quotes for the \n to be recognized as a newline character.

File: spec/csv_validator_spec.rb [hunk 10]
```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+require 'tempfile'
```
Comment (isaacanthony) [inline spec/csv_validator_spec.rb]: tempfile is already required in the spec_helper.

File: lib/filetype_validation/csv_validator.rb [hunk 11]
```diff
@@ -0,0 +1,40 @@
+require 'csv'
```
Comment (isaacanthony) [inline lib/filetype_validation/csv_validator.rb]: We should think about adding CSV as a dependency to this gem's gemspec.

File: filetype_validation.gemspec [hunk 12]
```diff
@@ -37,4 +37,5 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rake', '~> 10.0'
   spec.add_development_dependency 'rspec', '~> 3.0'
```
Comment (isaacanthony) [inline filetype_validation.gemspec]: Let's add minor versions to rake and rspec.

File: lib/filetype_validation/csv_validator.rb [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+    #
+    # This method checks to see if the file can have its
+    # first number_of_lines parsed as a csv file using the CSV
+    # library.
+    #
+    def parse_csv?
+      file_sample = File.foreach(file.path)
+                        .first(@options[:max_lines] || MAX_LINES)
+                        .join
+      CSV.parse(file_sample, headers: @options[:headers] || HEADERS)
```
Comment (isaacanthony) [inline lib/filetype_validation/csv_validator.rb]: Let's add just a quick write-up to the README Usage section where we provide quick docs for how to use the FiletypeValidation.csv?(file, options) API and what options are allowed.

File: spec/csv_validator_spec.rb (hunk 5 above)
Comment (isaacanthony) [inline spec/csv_validator_spec.rb]: This should really be FiletypeValidation::CsvValidator.new(file.flush, options).valid? so scoping our unit testing to methods in this file.

File: lib/filetype_validation/csv_validator.rb (hunk 11 above)
Comment (joejacob) [inline lib/filetype_validation/csv_validator.rb]: i've been googling, and it doesn't seem like it's required. i think it's meant mostly for gems that aren't in the standard lib

**Pull request merged**